hopsworks-api/src/main/java/io/hops/hopsworks/api/user/UserProfileBuilder.java

```diff
@@ -29,77 +31,132 @@
 import javax.ejb.Stateless;
 import javax.ejb.TransactionAttribute;
 import javax.ejb.TransactionAttributeType;
+import javax.servlet.http.HttpServletRequest;
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.List;
 import java.util.logging.Level;
+import java.util.logging.Logger;
 
 @Stateless
 @TransactionAttribute(TransactionAttributeType.NEVER)
 public class UserProfileBuilder {
+  private final static Logger LOGGER = Logger.getLogger(UserProfileBuilder.class.getName());
   @EJB
   private UsersController usersController;
   @EJB
   private UserFacade userFacade;
   @EJB
   private BbcGroupFacade bbcGroupFacade;
+  @EJB
+  private JWTHelper jwtHelper;
+  @EJB
+  private AccountAuditFacade accountAuditFacade;
 
-  public UserProfileDTO acceptUser(Integer userId, Users newUser) throws UserException, ServiceException {
+  private Users accept(Integer userId, Users newUser) throws UserException, ServiceException {
     Users u = usersController.getUserById(userId);
-    if (u.getStatus().equals(UserAccountStatus.VERIFIED_ACCOUNT)) {
-      Collection<BbcGroup> groups = null;
-      if (newUser != null) {
-        groups = newUser.getBbcGroupCollection();
-      }
-      if (groups == null || groups.isEmpty()) {
-        BbcGroup bbcGroup = bbcGroupFacade.findByGroupName("HOPS_USER");
-        groups = new ArrayList<>();
-        groups.add(bbcGroup);
+    Collection<BbcGroup> groups = null;
+    if (newUser != null) {
+      groups = newUser.getBbcGroupCollection();
+    }
+    if (groups == null || groups.isEmpty()) {
+      BbcGroup bbcGroup = bbcGroupFacade.findByGroupName("HOPS_USER");
+      groups = new ArrayList<>();
+      groups.add(bbcGroup);
+    }
+    u.setStatus(UserAccountStatus.ACTIVATED_ACCOUNT);
+    u.setBbcGroupCollection(groups);
+    u = userFacade.update(u);
+    usersController.sendConfirmationMail(u);
+    return u;
+  }
+
+  public UserProfileDTO acceptUser(Integer userId, Users newUser) throws UserException, ServiceException {
+    Users u = accept(userId, newUser);
+    return new UserProfileDTO(u);
+  }
+
+  public UserProfileDTO acceptUsers(List<Integer> ids, HttpServletRequest req) {
+    UserProfileDTO userProfileDTO = new UserProfileDTO();
+    Users target = null;
+    for (Integer id : ids) {
+      try {
+        target = accept(id, null);
+        userProfileDTO.addItem(new UserProfileDTO(target));
+      } catch (UserException | ServiceException e) {
+        LOGGER.log(Level.WARNING, "Failed to accept user with id: {0}", id);
```

Comment:
should there be no field in the userProfileDTO so that when we do bulk operation we can also specify the parts that failed? Here we just fail silently... and even return a OK back to the user/UI.

Reply:
There is no nice way to do that. I am returning the users that succeeded so the user knows if all users were accepted.